src/primitives.rs

```diff
@@ -113,13 +116,7 @@ pub mod rays {
             camera_transform: &GlobalTransform,
         ) -> Option<Self> {
             let view = camera_transform.compute_matrix();
-            let window = match windows.get(camera.window) {
-                Some(window) => window,
-                None => {
-                    error!("WindowId {} does not exist", camera.window);
-                    return None;
-                }
-            };
+            let window = get_window_for_camera(&windows, camera)?;

```

Comment:
`camera.window` was replaced by `camera.target`; this is an enum `RenderTarget` -- either a window or a texture:
```rust
pub enum RenderTarget {
    /// Window to which the camera's view is rendered.
    Window(WindowId),
    /// Image to which the camera's view is rendered.
    Image(Handle<Image>),
}
```

I refactored the match expressions that 1. either extract the window or 2. log an error and return `None` to a utility function, see below.